packages/sdk/src/subgraph/subgraph.ts

```diff
@@ -156,12 +156,11 @@ export class Subgraph {
   }
 
   async startPolling(): Promise<void> {
-    await Promise.all(
-      Object.keys(this.sdks).map(async (_chainId) => {
-        const chainId = parseInt(_chainId);
-        this.pollingStopperBlock[chainId] = await this.chainReader.getBlockNumber(chainId);
-      }),
-    );
+    Object.keys(this.sdks).map(async (_chainId) => {
```

Comment:
Isn't this a problem? This is going to be racing with the below code now.

Reply:
problem is coming cause lifi and xpollinate both initiate waitFor after initiating the sdk. And pollingLoop was undefined cause of promise
So this should work now.